### PR TITLE
Explain nuances of the term "container"

### PR DESCRIPTION
**Pull request by azarboon**
### Description

Marc Brooker has written [a blog post](https://brooker.co.za/blog/2023/06/19/container.html) explaining that "container" means different thing to different people, and that it should be checked whether audience and narrator use the same definition. Below are his four suggested definition. Given the authority of Kubernetes, I suggest this nuance to be mentioned. So readers are more informed and take cautionary acts to ensure they are synced with their audience. I've added it. 

> - An approach to packaging an application along with its dependencies (sometimes a whole operating system user space), that can then run on a minimal runtime environment with a clear contract.
> - A set of development, deployment, architectural, and operational approaches built around applications packaged this way.
> - A set of operational, security, and performance isolation tools that allow multiple applications to share an operating system without interfering with each other. On Linux, this tools include chroot, cgroups, namespaces, [seccomp](https://man7.org/linux/man-pages/man2/seccomp.2.html), and others.
> - A set of implementations of these practices (the proper nouns, Docker, Kubernetes, ECS, etc).

### Issue

Closes: #

### CODE REVIEW

**Comment by tengqm** [comment]
I'm okay with this note.
/lgtm

**Review by Arhell — COMMENTED** [review]
lgtm

**Comment by sftim** [comment]
/retitle Explain nuances of the term "container"

File: content/en/docs/concepts/containers/_index.md [hunk 1]
```diff
@@ -13,6 +13,8 @@ card:
 
 <!-- overview -->
 
+Note that "container" is an overloaded term. Whenever you use the word, check whether your audience uses the same definition.
```
Comment (sftim) [inline content/en/docs/concepts/containers/_index.md:16]: Maybe
```diff
-Note that "container" is an overloaded term. Whenever you use the word, check whether your audience uses the same definition.
+The word _container_ is an overloaded term. Whenever you use the word, check whether your audience uses the same definition.
```

But! I would put that framing in https://kubernetes.io/docs/contribute/style/style-guide/

In this introductory paragraph, I would an introductory explanation that the page is going to talk about _containers_ and _container images_, and will also explain a little about _containerization_ (using these
technologies in operations and during solution development).

Comment (azarboon) [reply content/en/docs/concepts/containers/_index.md]: Thanks for feedback. I presume you suggested that I add an introductory sentence so I tried to accommodate your feedbacks. 

This is my first PR and I'm not expert in Kubernetes. So, please feel free to edit work. I leave it up to you.

**Pull request merged**